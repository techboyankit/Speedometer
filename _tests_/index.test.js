/* eslint-disable react/forbid-foreign-prop-types */
import React from 'react';
import checkPropTypes from 'check-prop-types';
import TestRenderer from 'react-test-renderer';
import RNSpeedometer from '../src';

describe('Main Component Tests', () => {
  const value = 50;

  describe('check prop types', () => {
    it('should throw an error if value is not specified', () => {
      expect(
        checkPropTypes(
          RNSpeedometer.propTypes,
          {
            value: null,
          },
          'prop',
          RNSpeedometer.value,
        ),
      ).toBe('Failed prop type: The prop `value` is marked as required in `<<anonymous>>`, but its value is `null`.');
    });
  });

  describe('main component details', () => {
    const componentTestRenderer = TestRenderer.create(<RNSpeedometer
      value={value}
    />);

    const wrapper = componentTestRenderer.toJSON();
    it('should render the component correctly', () => {
      expect(componentTestRenderer).toMatchSnapshot();
    });

    it('should add the styles correctly', () => {
      expect(wrapper.props.style).toBeArray();
    });

    it('should contain a wrapper view ', () => {
      expect(wrapper.type).toBe('View');
    });

    it('should contain a array of children layouts ', () => {
      expect(wrapper.children).toBeArray();
    });
  });
});
