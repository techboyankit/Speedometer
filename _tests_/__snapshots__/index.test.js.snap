// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component Tests main component details should render the component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "marginVertical": 5,
      },
      Object {
        "height": 365,
        "width": 730,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#e6e6e6",
          "borderColor": "#ffffff",
          "justifyContent": "flex-end",
          "overflow": "hidden",
        },
        Object {
          "borderTopLeftRadius": 365,
          "borderTopRightRadius": 365,
          "height": 365,
          "width": 730,
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomRightRadius": 0,
            "borderTopRightRadius": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "backgroundColor": "#ff2900",
            "borderRadius": 365,
            "height": 730,
            "transform": Array [
              Object {
                "translateX": 182.5,
              },
              Object {
                "rotate": "90deg",
              },
              Object {
                "translateX": -182.5,
              },
            ],
            "width": 365,
          },
          Object {},
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderBottomRightRadius": 0,
            "borderTopRightRadius": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "backgroundColor": "#ff5400",
            "borderRadius": 365,
            "height": 730,
            "transform": Array [
              Object {
                "translateX": 182.5,
              },
              Object {
                "rotate": "120deg",
              },
              Object {
                "translateX": -182.5,
              },
            ],
            "width": 365,
          },
          Object {},
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderBottomRightRadius": 0,
            "borderTopRightRadius": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "backgroundColor": "#f4ab44",
            "borderRadius": 365,
            "height": 730,
            "transform": Array [
              Object {
                "translateX": 182.5,
              },
              Object {
                "rotate": "150deg",
              },
              Object {
                "translateX": -182.5,
              },
            ],
            "width": 365,
          },
          Object {},
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderBottomRightRadius": 0,
            "borderTopRightRadius": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "backgroundColor": "#f2cf1f",
            "borderRadius": 365,
            "height": 730,
            "transform": Array [
              Object {
                "translateX": 182.5,
              },
              Object {
                "rotate": "180deg",
              },
              Object {
                "translateX": -182.5,
              },
            ],
            "width": 365,
          },
          Object {},
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderBottomRightRadius": 0,
            "borderTopRightRadius": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "backgroundColor": "#14eb6e",
            "borderRadius": 365,
            "height": 730,
            "transform": Array [
              Object {
                "translateX": 182.5,
              },
              Object {
                "rotate": "210deg",
              },
              Object {
                "translateX": -182.5,
              },
            ],
            "width": 365,
          },
          Object {},
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "borderBottomRightRadius": 0,
            "borderTopRightRadius": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "backgroundColor": "#00ff6b",
            "borderRadius": 365,
            "height": 730,
            "transform": Array [
              Object {
                "translateX": 182.5,
              },
              Object {
                "rotate": "240deg",
              },
              Object {
                "translateX": -182.5,
              },
            ],
            "width": 365,
          },
          Object {},
        ]
      }
    />
    <View
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "top": -48.666666666666664,
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
          "zIndex": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/speedometer-needle.png",
          }
        }
        style={
          Array [
            Object {
              "height": 730,
              "resizeMode": "stretch",
              "width": 730,
            },
            Object {
              "height": 730,
              "width": 730,
            },
            Object {},
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderTopLeftRadius": 365,
            "borderTopRightRadius": 365,
            "height": 225,
            "justifyContent": "flex-end",
            "overflow": "hidden",
            "width": 450,
          },
          Object {
            "borderTopLeftRadius": 365,
            "borderTopRightRadius": 365,
            "height": 219,
            "width": 438,
          },
          Object {},
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "marginVertical": 5,
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 25,
            "fontWeight": "bold",
          },
          Object {},
        ]
      }
    >
      50
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {
            "color": "#f2cf1f",
          },
          Object {},
        ]
      }
    >
      Fair
    </Text>
  </View>
</View>
`;
